.github/workflows/gateway-contracts-upgrade-tests.yml

```diff
@@ -0,0 +1,176 @@
+name: gateway-contracts-upgrade-tests
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-changes:
+    name: gateway-contracts-upgrade-tests/check-changes
+    permissions:
+      actions: 'read'
+      contents: 'read'
+      pull-requests: 'read'
+    runs-on: ubuntu-latest
+    outputs:
+      changes-gw-contracts: ${{ steps.filter.outputs.gw-contracts }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: 'false'
+      - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36  # v3.0.2
+        id: filter
+        with:
+          filters: |
+            gw-contracts:
+              - .github/workflows/gateway-contracts-upgrade-tests.yml
+              - gateway-contracts/contracts/**
```

Comment:
Btw, I changed the filter to consider modifications on the contracts only, since I think if nothing has changed in the contracts then we shouldn't test any upgrade. Wdyt? @RomanBredehoft